examples/hosts/app-integration/external-data/src/model/taskList.ts

```diff
@@ -56,7 +56,13 @@ export class TaskList extends DataObject implements ITaskList {
      * collection pattern -- see the contact-collection example for more details on this pattern.
      */
     private readonly tasks = new Map<string, Task>();
-
+    private readonly draftData = new Map<string, Task>();
```

Comment:
Rather than making these in-memory maps (`new Map()`) we'll want them to be `SharedMap`s holding the persisted data.  This will then let us eliminate our usage of this.root to store the tasks.

To create the `SharedMap`s and start using them, you'll need to add the SharedMap factory to the TaskList factory's registry (it is already doing this for SharedCell/SharedString [here](https://github.com/microsoft/FluidFramework/blob/main/examples/hosts/app-integration/external-data/src/model/taskList.ts#L234)).  Then you'll want to create them on first initialization of the TaskList (similar to what CollaborativeTextarea does for its SharedString on first initialization [here](https://github.com/microsoft/FluidFramework/blob/main/examples/apps/collaborative-textarea/src/fluid-object/index.ts#L38)).

Both the draftData/savedData should store the `{ id, nameString, priorityCell }` type, since the `Task` object can't be serialized by the map.

LMK if you'd like to chat about it a bit, I'm pretty free this afternoon.